### PR TITLE
Ground work for switching to LWR server

### PR DESCRIPTION
**Pull request by maliroteh-sf**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -4,142 +4,162 @@
         {
             "type": "node",
             "request": "launch",
-            "name": "iOS - Preview",
```
Comment (maliroteh-sf) [inline .vscode/launch.json:7]: Not much changes to this file. I just changed the name of the entries to start with the command name first, and also reordered the entries to group the commands together.

File: src/common/LwrServerUtils.ts [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+        return lwrApp
+            .listen(() => {
+                // tslint:disable-next-line: no-console
+                console.log(
+                    `Listening on port ${runtimeConfig.port} ( mode = ${runtimeConfig.serverMode} , type = ${runtimeConfig.serverType} )`
+                );
+                LwrServerUtils.setServerIdleTimeout(lwrApp, idleTimeout, () => {
+                    if (exitOnIdle) {
+                        process.exit(0); // kill the process on server timeout
+                    }
```
Comment (maliroteh-sf) [inline src/common/LwrServerUtils.ts:44]: When running the product code we do want to exit the process but when running unit tests we don't want that to happen. The simplest/cleanest way to make this overridable was to add `exitOnIdle` parameter to `startLwrServer` function.

Comment (sfdctaka) [reply src/common/LwrServerUtils.ts:44]: Why do we want the process to keep running for unit test?

Comment (maliroteh-sf) [reply src/common/LwrServerUtils.ts:44]: B/c exiting the process will throw off jest test runner. Jest test runner will exit the test process by itself after the test is done running and if it does not have any handles/ports left open. If it does have handle/ports left open then jest will print out a warning about that (which is a great way to ensure that our processes clean up after themselves). The fact that our tests runs to the end and jest test running doesn't print out any warning at the end means that we are properly shutting down the server when we detect it being idle.

**Review by sfdctaka — APPROVED** [review]
LGTM! Ran unit tests locally and everything worked 👍

**Review by khawkins — APPROVED** [review]
Seems good to me! It's hard to know how much to dig into some of the details, since we're in uncertain territory with regards to their longevity. But it all makes sense and looks pretty well structured. 🙂 👍 :shipit:

**Pull request merged**